File: javaapp/src/main/java/com/wzx/javaapp/BindingAdapterRecyclerView.java
package com.wzx.javaapp;

import android.databinding.BindingAdapter;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

/**
 * 描述：
 * <p>
 * 创建人： Administrator
 * 创建时间： 2018/10/17
 * 更新时间：
 * 更新内容：
 */

public class BindingAdapterRecyclerView {

    @BindingAdapter({"adapter", "laoutManager"})
    public static void initRecycler(@NonNull RecyclerView recyclerView,
                                    @NonNull RecyclerView.Adapter<RecyclerView.ViewHolder> adapter,
                                    @Nullable RecyclerView.LayoutManager layoutManager) {
        recyclerView.setLayoutManager(layoutManager == null ? new LinearLayoutManager(recyclerView.getContext()) : layoutManager);
        recyclerView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/wzx/mvvmdemo/utils/BindingAdapterRecyclerView.kt
package com.wzx.mvvmdemo.utils

import android.databinding.BindingAdapter
import android.support.annotation.NonNull
import android.support.annotation.Nullable
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

/**
 * 描述：
 *
 * 创建人： Administrator
 * 创建时间： 2018/10/16
 * 更新时间：
 * 更新内容：
 */
class BindingAdapterRecyclerView {

    @BindingAdapter("adapter", "laoutManager")
    open fun initRecycler(@NonNull recyclerView: RecyclerView,
                     @NonNull adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>,
                     @Nullable layoutManager: RecyclerView.LayoutManager?) {
        recyclerView.layoutManager = layoutManager ?: LinearLayoutManager(recyclerView.context)
        recyclerView.adapter = adapter
    }
}